infrastructure/code/template/details.html

```diff
@@ -908,27 +944,102 @@ <h2 class="example-fixes-header">Fixed Code Examples</h2>
+     * Represents a commit fix with detailed change lines.
+     * @typedef {Object} ExampleCommitFix
+     * @property {string} CommitURL           - URL to the full commit diff.
+     * @property {string} RepoName            - Name of the repository.
+     * @property {string} RepoLink            - Direct link to the repository.
+     * @property {ExampleLine[]} ExampleLines - An array of line changes.
+     */
+
+    // Constants for class and ID names
     const MAIN_TAB_NAV_SELECTOR = '.main-tabs-nav';
     const MAIN_TAB_ITEM_SELECTOR = '.main-tabs-nav .tab-item';
     const MAIN_TAB_CONTENT_SELECTOR = '.tab-container > .main-tab-content';
-    const EXAMPLE_TAB_NAV_SELECTOR = '.example-fixes-tabs-nav';
-    const EXAMPLE_TAB_ITEM_SELECTOR = '.example-fixes-tabs-nav .tab-item';
-    const EXAMPLE_TAB_CONTENT_SELECTOR = '.tab-container > .example-fix-tab-content';
 
+    const EXAMPLE_COUNTER = 'example-counter';
+    const EXAMPLE_PREVIOUS = 'previous-example';
+    const EXAMPLE_NEXT = 'next-example';
+    const EXAMPLE_REPO_ANCHOR = 'example-repo-anchor';
+    const ALL_EXAMPLES_DETAIL = '.example-detail';
+
+    // Utility functions
     const toggleIsSelectedClass = (elements, shouldToggle) => {
       elements.forEach(el => {
         // toggle(token, force) - force is a boolean value that determines whether the class should be added or removed.
         el.classList.toggle('is-selected', shouldToggle(el));
       });
+    };
+
+    function toggleElement(element, toggle) {
+      if (!element) {
+        return;
+      }
+
+      if (toggle === 'show') {
+        element.classList.remove('hidden');
+      } else if (toggle === 'hide') {
+        element.classList.add('hidden');
+      } else {
+        console.error('Unexpected toggle value', toggle);
+      }
     }
 
-    // Ensure the document is fully loaded before executing script to manipulate DOM.
     document.addEventListener('DOMContentLoaded', () => {
+      const exampleCounter = document.getElementById(EXAMPLE_COUNTER);
+      const examplePrevious = document.getElementById(EXAMPLE_PREVIOUS);
+      const exampleNext = document.getElementById(EXAMPLE_NEXT);
+      const exampleRepoAnchor = document.getElementById(EXAMPLE_REPO_ANCHOR);
+      const allExamplesDetail = document.querySelectorAll(ALL_EXAMPLES_DETAIL);
       const mainTabLinks = document.querySelectorAll(MAIN_TAB_ITEM_SELECTOR);
       const mainTabContents = document.querySelectorAll(MAIN_TAB_CONTENT_SELECTOR);
-      // Nested tabs
-      const exampleTabLinks = document.querySelectorAll(EXAMPLE_TAB_ITEM_SELECTOR);
-      const exampleTabContents = document.querySelectorAll(EXAMPLE_TAB_CONTENT_SELECTOR);
+      const mainTabNav = document.querySelector(MAIN_TAB_NAV_SELECTOR);
+
+      let exampleCount = 0;
+
+      /** @type {ExampleCommitFix[]} */
+      const exampleCommitFixes = JSON.parse("{{.CommitFixes}}" || '[]');
+
+      // Functions dependent on the DOM
+      function showCurrentExample() {
```

Comment:
This is great! How about I raise a ticket to improve the code in here and `vscode-extension` by doing https://github.com/snyk/snyk-ls/pull/527#discussion_r1631390490? Basically we could expose these functions on the window and then both Example fixes and AI fixes will use the same functionality to move between tabs (in both VSCode and eventually IntelliJ).

Reply:
https://snyksec.atlassian.net/browse/IDE-409